### PR TITLE
Detector-Competition-Feat: Add PayTm Payment Detector

### PR DESCRIPTION
**Pull request by fumblehool**
### Description:
PR to add [PayTm ](https://paytm.com/) Payment Detector

API Docs: https://business.paytm.com/docs/api/initiate-transaction-api/

### Checklist:
* [ ] Tests passing (`make test-community`)?
* [X] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### Tests

Screenshot 2023-10-19 at 5 24 33 PM

### CODE REVIEW

**Comment by zricethezav** [comment]
@fumblehool same goes for this PR, could you provide a passing test screenshot?

**Comment by zricethezav** [comment]
@fumblehool would you mind fixing merge conflicts

**Comment by fumblehool** [comment]
@zricethezav Updated the PR, please re-review 😄

**Review by zricethezav — APPROVED** [review]

File: pkg/detectors/paytm/paytm.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+			Raw:          []byte(resMatch),
+		}
+
+		if verify {
+			client := s.client
+			if client == nil {
+				client = defaultClient
+			}
+
+			payload := strings.NewReader(`{"requestType":"Payment","mid":"Mid-1234","websiteName":"Sample-website-name","orderId":"ORDERID_1234","callbackUrl":"http://localhost:3000/paytm","txnAmount":{"value":"1.00","currenty":"INR"},"userInfo":{"custId":"customer-id-12345"}}`)
```
Comment (zricethezav) [inline pkg/detectors/paytm/paytm.go:57]: @fumblehool anyway we could have this not hit `https://securegw.paytm.in/theia/api/v1/initiateTransaction?`? Seems like there is a risk of sending funds

**Review by zricethezav — CHANGES_REQUESTED** [review]
revoking my approval because I have one question about the endpoint we're hitting

**Comment by zricethezav** [comment]
Closing this due to inactivity

**Pull request closed without merging**